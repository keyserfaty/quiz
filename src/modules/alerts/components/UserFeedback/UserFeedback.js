import React from 'react'
import Alert from '../../../../components/Alert'
import {TouchableOpacity, Text} from 'react-native'

import styles from './statics/styles'

const UserFeedback = props => {
  const { onCloseModal, data } = props
  
  return (
    <Alert type={data.type} title={data.title} subtitle={data.subtitle}>
      <TouchableOpacity style={styles.footer} onPress={onCloseModal}>
        <Text style={styles.buttonText}>Accept</Text>
      </TouchableOpacity>
    </Alert>
  )
}

export default UserFeedback
