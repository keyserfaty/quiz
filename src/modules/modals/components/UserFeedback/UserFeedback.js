import React from 'react'
import Modal from '../../../../components/Modal/Modal'
import {TouchableOpacity, Text} from 'react-native'

import styles from './statics/styles'

const UserFeedback = props => {
  const { onCloseModal, data } = props
  
  return (
    <Modal title={data.title} subtitle={data.subtitle}>
      <TouchableOpacity style={styles.footer} onPress={onCloseModal}>
        <Text style={styles.buttonText}>Accept</Text>
      </TouchableOpacity>
    </Modal>
  )
}

export default UserFeedback
