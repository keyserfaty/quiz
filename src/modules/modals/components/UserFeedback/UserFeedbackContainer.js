import React from 'react'
import UserFeedback from './UserFeedback'

class UserFeedbackContainer extends React.Component {
  render () {
    return (
      <UserFeedback {...this.props} />
    )
  }
}

export default UserFeedbackContainer
