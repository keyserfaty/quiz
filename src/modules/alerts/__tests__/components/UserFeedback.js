import React from 'react'
import UserFeedback from '../../components/UserFeedback/UserFeedback'

import Shallow from 'react-test-renderer/shallow'

const renderer = new Shallow()

test('Should render success', () => {
  const props = {
    data: {
      type: 'success',
      msg: 'some text'
    },
    onCloseAlert: () => {}
  }

  const component = renderer.render(<UserFeedback {...props} />)
  expect(component).toMatchSnapshot()
})

test('Should render failure', () => {
  const props = {
    data: {
      type: 'failure',
      msg: 'some text'
    },
    onCloseAlert: () => {}
  }

  const component = renderer.render(<UserFeedback {...props} />)
  expect(component).toMatchSnapshot()
})
