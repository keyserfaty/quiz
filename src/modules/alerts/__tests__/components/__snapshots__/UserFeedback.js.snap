// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render failure 1`] = `
<Alert
  msg="some text"
  onCloseAlert={[Function]}
  type="failure"
/>
`;

exports[`Should render success 1`] = `
<Alert
  msg="some text"
  onCloseAlert={[Function]}
  type="success"
/>
`;
